src/basis/router.js

```diff
@@ -256,41 +298,45 @@
       this.params = {};
 
       this.attach(function(values){
-        for (var key in this.paramsConfig_)
+        basis.object.iterate(this.paramsConfig_, function(key, paramConfig){
           if (values && key in values)
-            basis.Token.prototype.set.call(this.params[key], values[key]);
+            Value.prototype.set.call(this.params[key], values[key]);
           else
-            basis.Token.prototype.set.call(this.params[key], this.defaults_[key]);
+            Value.prototype.set.call(this.params[key], paramConfig.defaultValue);
+        }, this);
+
       }, this);
 
-      basis.object.iterate(this.paramsConfig_, function(key, transform){
-        var token = new basis.Token(this.defaults_[key]);
+      basis.object.iterate(this.paramsConfig_, function(key, paramConfig){
+        var paramValue = new Value({
+          value: paramConfig.defaultValue
+        });
 
-        token.set = function(value){
-          if (!this.matched.value)
+        paramValue.set = function(value){
```

Comment:
It can be moved to config. But `this` can't be replaced with `paramValue`, because `this` - is a route, not a value. I will rewrite this piece of code to make it more clear